arcaflow_plugin_horreum_client/horreum_client_plugin.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python3
+
+import sys
+import typing
+import requests
+import json
+
+from arcaflow_plugin_sdk import plugin
+from horreum_client_schema import InputParams, SuccessOutput, ErrorOutput
+
+
+@plugin.step(
+    id="horreum-client",
+    name="Horreum Client",
+    description="Uploads an object to the Horreum server",
+    outputs={"success": SuccessOutput, "error": ErrorOutput},
+)
+def horreum_client(
+    params: InputParams,
+) -> typing.Tuple[str, typing.Union[SuccessOutput, ErrorOutput]]:
+    auth_url = (
+        params.horreum_keycloak_url + "/realms/horreum/protocol/openid-connect/token"
+    )
+    auth_obj = {
+        "username": params.horreum_username,
+        "password": params.horreum_password,
+        "grant_type": "password",
+        "client_id": "horreum-ui",
+    }
+
+    try:
+        print("==> Authenticating with keycloak...")
+        auth_return = requests.post(auth_url, data=auth_obj, verify=False)
+        token = json.loads(auth_return.text)["access_token"]
+
+    except (requests.ConnectionError, requests.HTTPError, requests.Timeout):
+        return "error", ErrorOutput(
+            f"Error communicating with server: {auth_return.text}"
+        )
+
+    send_url = (
+        params.horreum_url
+        + f"/api/run/data?test={params.test_name}&start={params.test_start_time}"
+        f"&stop={params.test_stop_time}&owner={params.test_owner}"
+        f"&access={params.test_access_rights}"
```

Comment:
It's slightly odd to use both string concatenation and f-strings.  I would stick with one:
```python
        f"{params.horreum_url}/api/run/data?test={params.test_name}"
        f"&start={params.test_start_time}&stop={params.test_stop_time}"
        f"&owner={params.test_owner}&access={params.test_access_rights}"
```

Reply:
Cleaned up.